java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt

```diff
@@ -4681,7 +4687,7 @@ open class KotlinFileExtractor(
         addVisibilityModifierToLocalOrAnonymousClass(id)
         extractClassSupertypes(superTypes, listOf(), id, inReceiverContext = true)
 
-        extractEnclosingClass(currentDeclaration, id, locId, listOf())
+        extractEnclosingClass(declarationParent, id, null, locId, listOf())
```

Comment:
Should the `null` be `currentDeclaration as? IrClass`?

Reply:
I think you mean `elementToReportOn as? IrClass`, but that will always be `null`, because the element is either a function reference, a property reference, a local function or an `IrTypeOperatorCall`.